app/javascript/react/components/NewReviewContainer.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react'
+import { Redirect } from 'react-router-dom'
+
+import NewReviewForm from './NewReviewForm'
+
+const NewReviewContainer = props => {
+  const game_id = props.game_id
+  const [shouldRedirect, setShouldRedirect] = useState(false)
+
+  const handleFormSubmit = formData => {
+    formData.game_id = game_id
+    fetch('/api/v1/reviews', {
+      credentials: "same-origin",
+      method: "POST",
+      body: JSON.stringify(formData),
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json"
+      }
+    })
+    .then(response => {
+      if(response.ok) {
+        return response
+      } else {
+        let errorMessage = `${response.status} (${response.statusText})`,
+        error = new Error(errorMessage)
+        throw error
+      }
+    })
+    .then(response => response.json())
+    .then(parsedData => {
+      if (parsedData.errors){
+      setErrors(parsedData.errors)
+    }
+      setShouldRedirect(true)
+    })
```

Comment:
lines 32 through 35 need to be rethought. 

currently, submitting the form will ALWAYS redirect the page via react router, which makes lines 32 through 34 defunct. It should be if...else